tests/functional/generate_pages.py

```diff
@@ -392,78 +392,71 @@ def process_calculated_summary(answers, page_spec):
-                        ),
-                    }
-                    for answer in question.get("answers", []):
-                        answer_name = generate_pascal_case_from_id(answer["id"])
-                        answer_context = {
-                            "answerName": camel_case(answer_name),
-                            "answerId": answer["id"],
-                        }
-                        if section_summary:
-                            page_spec.write(
-                                SECTION_SUMMARY_ANSWER_GETTER.substitute(answer_context)
-                            )
-                            page_spec.write(
-                                SECTION_SUMMARY_ANSWER_EDIT_GETTER.substitute(
-                                    answer_context
-                                )
-                            )
+    if collapsible:
+        page_spec.write(COLLAPSIBLE_SUMMARY_GETTER)
 
+
+def write_summary_spec(collapsible, page_spec, section, section_summary):
+    list_summaries = [
+        summary_element
+        for summary_element in section.get("summary", {}).get("items", [])
+        if summary_element["type"] == "List"
+    ]
+    for list_block in list_summaries:
+        list_context = {"list_name": list_block["for_list"]}
+        page_spec.write(LIST_SECTION_SUMMARY_ADD_LINK_GETTER.substitute(list_context))
+        page_spec.write(LIST_SECTION_SUMMARY_EDIT_LINK_GETTER.substitute(list_context))
+        page_spec.write(
+            LIST_SECTION_SUMMARY_REMOVE_LINK_GETTER.substitute(list_context)
+        )
+        page_spec.write(LIST_SECTION_SUMMARY_LABEL_GETTER.substitute(list_context))
+
+    for group in section["groups"]:
+        for block in group["blocks"]:
+            for question in get_all_questions(block):
+                question_context = {
+                    "questionId": question["id"],
+                    "questionName": camel_case(
+                        generate_pascal_case_from_id(question["id"])
+                    ),
+                }
+                for answer in question.get("answers", []):
+                    answer_name = generate_pascal_case_from_id(answer["id"])
+                    answer_context = {
+                        "answerName": camel_case(answer_name),
+                        "answerId": answer["id"],
+                    }
+                    if section_summary:
                         page_spec.write(
-                            SUMMARY_ANSWER_GETTER.substitute(answer_context)
+                            SECTION_SUMMARY_ANSWER_GETTER.substitute(answer_context)
                         )
-
                         page_spec.write(
-                            SUMMARY_ANSWER_EDIT_GETTER.substitute(answer_context)
+                            SECTION_SUMMARY_ANSWER_EDIT_GETTER.substitute(
+                                answer_context
+                            )
                         )
```

Comment:
I think we can drop this `if` block as it generates the same accessors as the `SUMMARY_ANSWER_GETTER` and

Reply:
We will call that 1 on the refactoring card  as SECTION_SUMMARY_ANSWER_GETTER etc wasn't added by this story